clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitRequest.java

```diff
@@ -191,18 +213,12 @@ public Builder setGenerationId(int generationId) {
 
         @Override
         public OffsetCommitRequest build(short version) {
-            switch (version) {
-                case 0:
-                    return new OffsetCommitRequest(groupId, DEFAULT_GENERATION_ID, DEFAULT_MEMBER_ID,
-                            DEFAULT_RETENTION_TIME, offsetData, version);
-                case 1:
-                case 2:
-                case 3:
-                case 4:
-                case 5:
-                    return new OffsetCommitRequest(groupId, generationId, memberId, DEFAULT_RETENTION_TIME, offsetData, version);
-                default:
-                    throw new UnsupportedVersionException("Unsupported version " + version);
+            if (version == 0) {
```

Comment:
It seems that the previous approach is a bit better and stricter because it throws `UnsupportedVersionException` for versions that are not supported. Also it may be good to use the same approach as build() for other requests for consistency.

Reply:
There is some inconsistency between request types. This is the only request I found that had all versions explicitly listed. I think we can probably move the version check into `AbstractRequest` to make the code friendlier to maintain while still ensuring the strict version validation.